### PR TITLE
Fixing Swagger KPIs and RP request response payloads

### PR DESCRIPTION
**Pull request by harshivcodes**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify: addressing feedback from Tooling teams for better customer experience.
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by harshitha-akkaraju** [comment]
@heaths @RandalliLama @jlichwa 

We're still in private-preview and no SDK releases have taken place. And the fields that have been removed also don't impact customers. Please let me know if you'd like me to schedule time on your calendars for an ARM review (no new APIs have been added; only contract changes and swagger refactoring).

**Review by heaths — APPROVED** [review]
Formatting changes to Key Vault files (3) look fine. I can only sign off on those.

**Comment by j5lim** [comment]
> @heaths @RandalliLama @jlichwa
> 
> We're still in private-preview and no SDK releases have taken place. And the fields that have been removed also don't impact customers. Please let me know if you'd like me to schedule time on your calendars for an ARM review (no new APIs have been added; only contract changes and swagger refactoring).

@harshitha-akkaraju Please follow the steps in "Breaking Change Review Checklist" and request for an approval with justification.

**Comment by heaths** [comment]
@jlichwa @RandalliLama and I are only codeowners for Key Vault, which seems these mere *samples* were updated to fix a "prettier" issue. I'm not sure a review with us for Networking or Storage Pool APIs is warranted.

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (jorgecotillo) [inline specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:917]: I think provisioningState and status are set by the service and are not set by a user, consider 2 things: 1. Marking properties set by the service as readOnly: true and 2. Removing them from "required"

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (jorgecotillo) [inline specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:990]: if you max is 8, consider adding it in the description

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (jorgecotillo) [inline specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:1025]: Same as above, add the max length as part of the description

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (jorgecotillo) [inline specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:1127]: Make sure to add to the required list those properties that the user must set, unless there's a different motivation to add these (ProvisioningState and Status) under required?

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (jorgecotillo) [inline specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:1179]: Just a list of 1? Make sure to let the user know by adding the constraint in the description.

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (jorgecotillo) [inline specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:1382]: Mark this property as a secret, see here:

https://armwiki.azurewebsites.net/rp_onboarding/process/property_design_best_practices.html?q=x-ms-secret

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (jorgecotillo) [inline specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:1404]: Please review your property constraints and add them into the description, this way is clear for end users about the constraints they'll face.

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (harshitha-akkaraju) [reply specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:917]: The customer should be using DiskPoolCreate or DiskPoolUpdate to write to the resource and not DiskPool. We did this refactoring in response to feedback from one of the tooling teams.

The RP uses the DiskPool object as part of the response payloads for GET, PUT, PATCH requests on a Disk pool. I've marked the provisioningState and status as required -- to communicate that the RP is responsible for populating all these fields as part of the response payload.

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (harshitha-akkaraju) [reply specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:1127]: Yeah -- same as above. We use IscsiTarget and the nested properties IscsiTargetProperties as the response payload returned by the RP. So the customers shouldn't be using it to make create/update requests.

They use IscsiTargetCreate/IscsiTargetUpdate for PUT/PATCH requests on the resource

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (harshitha-akkaraju) [reply specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:1382]: We mark the entire object as a secret on this [line](https://github.com/Azure/azure-rest-api-specs/blob/5acaa21b6511c0e43e5f9e6c45da50437a6dca51/specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json#L1338), will that do?

File: specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json
Comment (harshitha-akkaraju) [reply specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json:1404]: corrected the constraints - thanks!

**Review by heaths — APPROVED** [review]

**Review by pilor — APPROVED** [review]
Approving with breaking changes due to the s360 flag on the PR

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**
